From 181041705657126c1db35e994399d991230575f5 Mon Sep 17 00:00:00 2001
From: Florent Kermarrec <florent@enjoy-digital.fr>
Date: Tue, 9 Sep 2025 09:54:51 +0200
Subject: [PATCH] build/vhd2v_converter: Avoid mutable defaults by using None
 as the default and creating fresh lists inside __init__.

Also print main ghdl command to ease debug.
---
 litex/build/vhd2v_converter.py | 9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)

diff --git a/litex/build/vhd2v_converter.py b/litex/build/vhd2v_converter.py
index a054e3aac..2af24aa57 100644
--- a/litex/build/vhd2v_converter.py
+++ b/litex/build/vhd2v_converter.py
@@ -59,11 +59,15 @@ class VHD2VConverter(Module):
         add_instance   = False,
         params         = None,
         instance       = None,
-        files          = list(),
-        libraries      = list()):
+        files          = None,
+        libraries      = None):
         """
         constructor (see class attributes)
         """
+        if files is None:
+            files = []
+        if libraries is None:
+            libraries = []
         self._top_entity     = top_entity
         self._build_dir      = build_dir
         self._work_package   = work_package
@@ -204,6 +208,7 @@ class VHD2VConverter(Module):
             cmd += generics
             cmd += self._sources
             cmd += ["-e", self._top_entity]
+            print(cmd)
 
             with open(verilog_out, 'w') as output:
                 s = subprocess.run(cmd, stdout=output)
-- 
2.51.0

